### ISSUE TITLE
Poor video quality

### ISSUE THREAD

**Issue by Pepo360** [signalapp/Signal-iOS#4370]
Hello.
Firstly, I have to say that I really appreciate all the amazing work development team is doing. There is a lot of effort and hard work behind this astonishing project. However, there seems to be a huge issue with compression of video files from iPhone photo library. Quality of video after compression is really bad. I can barely read the text when a friend send me screen recording from his iPhone. After doing some research on the internet, this issue seems to be here for a while. As far as I know, the huge loss in quality only happens on IOS, when the MOV video file is being converted to mp4 file. When I recieve a video sent from an android phone, quality seems just fine.

**Comment by leifmarcus** [comment]
Would be nice if the video quality would be better for videos send via iOS

**Comment by cabiamdos** [comment]
It happens me exactly this issue. Signal downgrade a lot the quality when I sent video and photos from my phone to other people phones.

I am super confused.

**Comment by kaktusztea** [comment]
Yes, this is still an issue. Video quality is ruined if you send it from an iDevice.

**Comment by kaktusztea** [comment]
Yes, this is one of the most relevant issue at the moment.

**Comment by k3rne1pan1c** [comment]
Is this something we can fix by ourselves? Is this a problem of the IOS/Android video implementation or does Signal have their own servers to then minimize the server load?

**Comment by NiklasBr** [comment]
I think this blog post sheds some light on the architecture: https://signal.org/blog/how-to-build-encrypted-group-calls/

**Comment by ianrahman** [comment]
The video sharing functionality appears entirely unrelated to group video calls.

**Comment by AndrewKvalheim** [comment]
> Cloudflare limits uploads to 100 MB

Even cuts of just a couple seconds are degraded to the point of being unusable.

**Comment by kaktusztea** [comment]
Still an issue.

**Comment by ianrahman** [comment]
Please reopen this issue as it continues to persist. Thank you.

**Comment by ianrahman** [comment]
Bump

**Comment by jj-wohlgemuth** [comment]
It's an issue for if you post videos in stories as well.

**Comment by ianrahman** [comment]
Bump

**Comment by bluekeybo** [comment]
How is this still an issue in 2024, it's ridiculous. Please improve image/video/call quality

**Comment by alextsagkas** [comment]
The app is so great that I am thinking of switching completely to it. But this feature missing is a major reason that holds me back. Please make some progress to it!

**Comment by kaktusztea** [comment]
Not stale at all.

**Comment by musterknabe** [comment]
1980 called and wants it's VHS tapes back.

**Comment by frigorific** [comment]
Yes, finally got family on Signal and suddenly found out about the 360p video attachment from iOS devices. Dosent change with high setting. 

How videos are encoded:
IOS 360p main profile 3 - 2.5 mbit/s
Android 720p high profile 5.2 - 2.5 mbit/s

Even the 720p "hd" on Android needs a bump to 1080p and either a bit higher bitrate or more efficient encode. Can not send video as file without zipping it either..

Other than that, a lot of top notch work going on right now! 😊😊❤️

![1000055643](https://github.com/user-attachments/assets/d01e577a-eca4-417c-9c4a-2bc09ae87191)
![1000055640](https://github.com/user-attachments/assets/fa6741f3-ae52-44eb-829f-794f6bf344d2)

**Comment by jacobrmoss** [comment]
Yeah I'm dealing with this same issue. Not stale at all, and very confusing.

**Comment by iaxat** [comment]
The poor video quality is also during video calls, I have been on Signal since such a long time, but this issue persists.

**Comment by iaxat** [comment]
Right?

I agree, it's been so long, doing video calls via signal is a no-brainer, you can't even see the faces clearly. And at the same time, any other app is much better.

**Comment by AlecKinnear** [comment]
Why is it necessary to use Cloudflare infrastructure? I'd be fine with some kind of paid tier with better storage but where payment is not attached to the actual calls. I'd just to have clean calls without falling back to Skype or Facetime (zero security, easily recorded by default by security services).

Signal already knows who we are by the obligatory phone number, that information and our networks are inadvertently shared with intelligence agencies, who can map out the frequency of our calls and our correspondents but probably can not get the actual content via Signal (would have to do that via the device). Anyway since Signal knows who we are and gives that away, payment shouldn't be an issue of identity.

**Comment by wassim-devel** [comment]
I have the same issue. It really should be fixed

**Comment by Stensel8** [comment]
Still having issues with low quality photo and video. Even after enabling HD media.

Using S24 Ultra running OneUI7/Android15.

**Comment by sweetbabyjava** [comment]
Still having issues with low quality photo and video. Even after enabling HD media.

iPhone 14 iOS 18.5

**Comment by Falsen** [comment]
Any updates? I still have this problem on my iPhone 13

**Comment by hamp007** [comment]
Yeah, I'm about tired of the "it's an android" comments for this app. RCS does way better than this.

**Comment by ianrahman** [comment]
PR with potential fix: https://github.com/signalapp/Signal-iOS/pull/6136

Turns out it isn't just the 100 MB Cloudflare limit, but also a blunt down-res of attached videos using a 640x480 preset instead of respecting the original video resolution.

**Comment by wassim-devel** [comment]
Thank you for this PR! I hope it will get merged

**Comment by jcforbes57** [comment]
Another user here frustrated by the video quality. I tried to share a 15 second screen capture of navigating a website on my Samsung S24U and the text, very large headline sized text, is completely illegible. I read the dev reply to the PR and there's some hope I suppose. ![Screenshot_20251227_132353_Signal.jpg](https://github.com/user-attachments/assets/81cb6bf0-04c6-4efe-abb8-51246ff1cbfd)

**Comment by AlecKinnear** [comment]
Sadly we've given up on Signal video calls in our family. The video quality is too inconsistent. Telegram has been an excellent substitute with very good video quality and excellent audio quality all the time. Most of Telegram is open source as well, so perhaps the Signal developers could get off their duff and copy the Telegram video and audio call protocols. Signal audio calls have been usable so existing audio call code is not the same write-off which the video call protocols are.

**Comment by iaxat** [comment]
I think even Zoom is better than Telegram, the amount of controversy and shadiness Telegram has been involved in.

**Comment by AlecKinnear** [comment]
Security and privacy is much better on Telegram than on Zoom or WhatsApp. I hope your reasoning Kumar here goes beyond "Russia bad, Durov born in Russia, Telegram bad".

In terms of security and privacy, I'd recommend being extremely careful with what you write or say on Signal or Telegram. With WhatsApp and Zoom you can assume the calls and transcripts and texts are saved forever in a central NSA database just like any Facebook or Google-based communication. For most of us, Russia or China are far less of a threat than the US, Five Eyes or the EU. Chairman Xi does not care about your real estate business in Portland or your local school board or whom you support for congressman/congresswoman or the interview you gave to the Portland Tribune about corruption with zoning permits.

In any case, Telegram video calls work perfectly, whether nearby or transatlantic. Signal video calls are so bad that we've given them up completely in our family. Signal to which I enlisted most of my family, in the end has done a great deal to keep us apart as call quality is so bad we had to give up video calls entirely.

Returning to Signal video quality: let's get some better minds on the video protocol and rebuild it. What's in there now simple does not work. It seems to me ten years ago or so there was a working protocol as when we started with Signal we were able to enjoy slightly better video calls.

Back then Facetime was much better. We had expected Signal would get better and occasionally it did or maybe it was just luck or exactly the right combination of OS (including versions).

**What can we collectively do to fix Signal video call quality?**

**Comment by wassim-devel** [comment]
> Security and privacy is much better on Telegram than on Zoom or WhatsApp.

Maybe better than Zoom, but clearly not better than WhatsApp. Telegram does not have E2EE for messages by default while on WhatsApp E2EE is mandatory

**Comment by AlecKinnear** [comment]
You should [read the news more often](https://tbot.substack.com/p/meta-whistleblower-engineers-access-whatsapp-data).

Long before this lawsuit, the lack of security and E2EE was clear. Just from the public revelations over the last ten years, it's been obvious that WhatsApp has always been leaky (many inconvenient European politicians have lost their mandate due to WhatsApp revelations). There are far fewer Telegram leaks.

The unknown if it's due to Telegram not keeping all the data or just that 1. Telegram security is better, with encryption for most states including rest 2. the Telegram team is smaller and more trustworthy than the 1500+ contractors who had WhatsApp access.

There's a footrace between Signal and Telegram over security (Signal leaks your connections and your frequency of communication but does not leak the actual content; Telegram has the data but does not leak it by default only under court order) but both are thousands of miles ahead of Zoom and WhatsApp and even Apple who leak all your data all the time as they are US-based and subject to the Patriot Act.

Since Durov and his girlfriend were held hostage by the French in Paris for nine months (from 24 August 2024 to 15 March 2025), Telegram is less secure. The French arrest was to force Telegram to install some external backdoors and monitoring into Telegram, where there had been none before. How much user privacy did Durov give up? It's not public knowledge yet. Today one can assume that with a court order, any significant Western government can obtain an individual's Telegram communication. Proton Mail is no better. Proton's track record is to give up communication when [faced with a French/Swiss court order](https://www.wired.com/story/protonmail-amends-policy-after-giving-up-activists-data/).

Signal cannot leak what it doesn't have (which is the actual content of the communication, the network they do have and you should assume Signal does leak the network activity: due to Patriot Act/US domicile/US citizens/past CIA money, Signal have to give up whatever they have).

**Which is why it's so very important that Signal video and audio call quality improve. Signal is the only messenger app where it's _likely_ that the interlocutors enjoy real privacy during calls.**

**Comment by wassim-devel** [comment]
Still an issue

**Comment by frigorific** [comment]
Yes, still an issue. I can see people are mixing in thoughts on video calling quality in this issue. I have no input on that, feels  good enough on my end. The video attachment has the same iOS bug still though. 

Tested again now. Found a ~12 second camera sourced video sent from Android and iOS. Both files when downloaded in app approx 4 MB filesize. Both about the same video bitrate ~2500 kb/s. But then the difference between iOS upload and Android upload. Attaching MediaInfo for files. Main profile 640 width encode from iOS. High profile 1280 width from Android.

Again, thank you all for making Signal super great the last few years ❤️ This bug will probably be sorted at some point.

<img width="1080" height="2280" alt="Image" src="https://github.com/user-attachments/assets/c6fedf00-4673-4574-812a-8f93ac91fd2e" />
<img width="1080" height="2286" alt="Image" src="https://github.com/user-attachments/assets/03a97d86-9503-4fdc-9095-19c596a46df4" />

**Comment by Stensel8** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

Still an issue, however it is not as bad as it used to be.

Quality of screenrecordings is meh, but text is not completely unreadable anymore.

Why don't we use AV1 / AVIF for media compression. Isn't that the newest and most efficient compression library?

I don't know how heavy it is for phones though. Maybe it is too heavy to compress/transcode.

**Comment by iaxat** [comment]
And video quality is poor for both iOS and Android, its just worse on iOS.

**Comment by mitomenai** [comment]
> Still an issue, however it is not as bad as it used to be.

I politely disagree. Videos are still being compressed to 360p. It's as bad today as it was in 2020. 

Choosing "High" in global settings has no impact on video size. Still being sent in 360p. Seems to be completely broken on iOS.

**Comment by Stensel8** [comment]
> > Still an issue, however it is not as bad as it used to be.
> 
> I politely disagree. Videos are still being compressed to 360p. It's as bad today as it was in 2020.
> 
> Choosing "High" in global settings has no impact on video size. Still being sent in 360p. Seems to be completely broken on iOS.

Sorry, my bad. I don't have an iPhone, but use an S24Ultra. I replied to the wrong repository.

On the Android side it seems to be a bit better now.

**Comment by ianrahman** [comment]
@max-signal are there any updates on this front?

**Comment by wassim-devel** [comment]
This issue really should be a main priority, I don't understand why so many features are added and issues fixed but this critical issue on an app used by millions of people still isn't solved after so many time, as if it was normal to get 360p videos

**Comment by kaktusztea** [comment]
It's still an issue.
